rocm-docs/caffe2-build.md

```diff
@@ -0,0 +1,89 @@
+# Caffe2: Building From Source on ROCm Platform
+
+## Intro
+This instruction provides a starting point to build caffe2 on AMD GPUs (Caffe2 ROCm port) from source.
+*Note*: it is recommended to start with a clean Ubuntu 16.04 system
+
+## Install ROCm
+
+Install ROCm stack following steps at [link](https://github.com/RadeonOpenCompute/ROCm/blob/master/README.md) if your machine doesn't have ROCm already.
+
+## Build caffe2 inside docker
+
+ If your machine doesn't have docker installed, follow the steps [here](https://docs.docker.com/install/linux/docker-ce/ubuntu/#install-docker-ce) to install docker.
+
+### Pull the docker image
+```
+docker pull rocm/caffe2:unbuilt-rocm1.8.2
+```
+This docker image has all the dependencies for caffe2 pre-installed.
+
+### Pull the latest caffe2 source:
+* Using https 
+```
+git clone --recurse-submodules https://github.com/ROCmSoftwarePlatform/pytorch.git
+```
+* Using ssh
+```
+git clone --recurse-submodules git@github.com:ROCmSoftwarePlatform/pytorch.git
+```
+Navigate to repo directory
+```
+cd pytorch
+```
+
+### Launch the docker container
+```	
+docker run -it --network=host --device=/dev/kfd --device=/dev/dri --group-add video -v $PWD:/pytorch rocm/caffe2:unbuilt-rocm1.8.2
+``` 
+Navigate to pytorch directory `cd /pytorch` inside the container.
+
+### Build caffe2 Project from Src
+
+* Run the command  
+
+	`.jenkins/caffe2/build.sh`
+
+	
+* Test the rocm-caffe2 Installation 
+
+	Before running the tests, make sure that the required environment variables are set:
+	``` 
+	export LD_LIBRARY_PATH=/usr/local/caffe2/lib:$LD_LIBRARY_PATH
+	export PYTHONPATH=/usr/local/caffe2/lib/python2.7/dist-packages:$PYTHONPATH
+	```
+
+	Run the binaries under `/pytorch/build_caffe2/bin`
+
+### Run benchmarks
+
+Navigate to build directory, `cd /pytorch/build_caffe2` to run benchmarks.
```

Comment:
Do these work on upstream C2 by default or do we need port them to AMD?

Reply:
They currently will work on rocm fork once this PR gets merged. We can make it work on upstream pytorch repo with changes to the Python API. I created a PR with those changes, but they do not want to make changes on python side until c++ runtime becomes stable.